packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/AddressFieldInput.tsx

```diff
@@ -0,0 +1,71 @@
+import { useAddressField } from '@/object-record/record-field/meta-types/hooks/useAddressField';
+import { FieldAddressDraftValue } from '@/object-record/record-field/types/FieldInputDraftValue';
+import { AddressInput } from '@/ui/field/input/components/AddressInput';
+import { FieldInputOverlay } from '@/ui/field/input/components/FieldInputOverlay';
+
+import { usePersistField } from '../../../hooks/usePersistField';
+
+import { FieldInputEvent } from './DateFieldInput';
+
+export type AddressFieldInputProps = {
+  onClickOutside?: FieldInputEvent;
+  onEnter?: FieldInputEvent;
+  onEscape?: FieldInputEvent;
+};
+
+export const AddressFieldInput = ({
+  onEnter,
+  onEscape,
+  onClickOutside,
+}: AddressFieldInputProps) => {
+  const { hotkeyScope, draftValue, setDraftValue } = useAddressField();
+
+  const persistField = usePersistField();
+
+  const convertToAddress = (
+    newAddress: FieldAddressDraftValue | undefined,
+  ): FieldAddressDraftValue => {
+    return {
+      addressStreet1: newAddress?.addressStreet1 ?? '',
+      addressStreet2: newAddress?.addressStreet2 ?? null,
+      addressCity: newAddress?.addressCity ?? null,
+      addressState: newAddress?.addressState ?? null,
+      addressCountry: newAddress?.addressCountry ?? null,
+      addressPostcode: newAddress?.addressPostcode ?? null,
+      addressLat: newAddress?.addressLat ?? null,
+      addressLng: newAddress?.addressLng ?? null,
+    };
+  };
```

Comment:
Not sure if this is a good way how to handle the default values. Id love to improve this if anyone know how.

Reply:
cc @lucasbordeau